susemanager-utils/susemanager-sls/modules/engines/mgr_events.py

```diff
@@ -127,13 +128,16 @@ def _insert(self, tag, data):
             fnmatch.fnmatch(tag, "suse/manager/image_deployed"),
             fnmatch.fnmatch(tag, "suse/systemid/generate")
         ]) and not self._is_salt_mine_event(tag, data) and not self._is_presence_ping(tag, data):
-            log.debug("%s: Adding event to queue -> %s", __name__, tag)
+            queue = int(hashlib.md5(data.get("id").encode(self.connection.encoding)).hexdigest()[0:8], 16) \
```

Comment:
I wonder if it is really useful to replicate the hashing in Postgres given we already have it here.

Would it work if we saved the queue number as an extra column in `suseSaltEvents` and then, on the Java side, just pop events from a certain queue?

The advantage would be that the hashing logic lives in just one place.

What do you think?

Reply:
A bit OT: everytime when I see the usage of MD5 I remember that this code will stop working as soon as somebody enable FIPS mode. We might not support this, but there might be a time comming where something like this is requested.

In such a case try to develop something where it is very easy to change the hashing algorithm to something else. And in best case you can do it at one place.

Reply:
@moio I'ld be OK to add such a column: indeed that would simplify the whole thing and avoid some queue number computing cycles.

@mcalmer I wasn't aware that md5 is not FIPS-compatible. The good thing is that if we add a queue column, the hash would be performed on the python side and we have way more possible algorithms at hand. I don't really want md5... it's just that it's the only one built in postgresql. Hopefully I could default to sha256 and add it as a setting for the engine.

Reply:
By the way this is not security-related, so a much less secure hash algorithm such as CRC or a specialized hash for data structures like hashbrown or SwissTable would also work.

https://blog.waffles.space/2018/12/07/deep-dive-into-hashbrown/

Still we are all on the same page - all things being equal one place is better than two.

Reply:
@moio do we really want to use an exotic hash algorithm and add some more python dependencies? python's hashlib guarantees `md5` and the `sha*` algorithms.

Reply:
The simplest option would be to use `binascii.crc32()` which seems to be shipped by default and is already on my test server instance

Reply:
Sure, pick the simplest option available in python's standard library

Reply:
Since we decided that md5 is OK as for non-security purpose and the queue is now computed in one place, you can resume the review